### ISSUE TITLE
setClip incorrectly restores prior font, paint, color, and stroke

### ISSUE THREAD

**Issue by turnef** [jfree/fxgraphics2d#6]
The setClip() method, when invoked after font, paint, color or stroke details are altered, incorrectly restores the prior state.  The following code demonstrates the issue:

```java
import java.awt.BasicStroke;
import java.awt.Color;
import java.awt.Rectangle;

import org.jfree.fx.FXGraphics2D;

import javafx.application.Application;
import javafx.scene.Group;
import javafx.scene.Scene;
import javafx.scene.canvas.Canvas;
import javafx.scene.canvas.GraphicsContext;
import javafx.stage.Stage;

/**
 * Simple code example demonstrating the issue with setClip()
 */
public class SetClipIssue extends Application {

	public static void main(String[] args) {
		launch(args);
	}

	@Override
	public void start(Stage primaryStage) throws Exception {
		primaryStage.setTitle("SetClipIssue");
		Group root = new Group();
		Canvas canvas = new Canvas(640, 480);
		GraphicsContext gc = canvas.getGraphicsContext2D();

		FXGraphics2D g2d = new FXGraphics2D(gc);
		g2d.setFont(g2d.getFont().deriveFont(20f));
		g2d.setColor(Color.RED);
		g2d.setStroke(new BasicStroke(2.0f));
		g2d.setClip(new Rectangle(50, 50, 540, 380));
		g2d.drawString("Red 20pt", 100, 100);
		g2d.draw(new Rectangle(55, 75, 25, 25));

		g2d.setFont(g2d.getFont().deriveFont(30f));
		g2d.setColor(Color.BLUE);
		g2d.setStroke(new BasicStroke(1.0f));

		/*
		 * This clip set incorrectly restores the previous state of color, font, and
		 * stroke.
		 */
		g2d.setClip(new Rectangle(49, 49, 542, 382));
		g2d.drawString("Blue 30pt?", 100, 200);
		g2d.draw(new Rectangle(55, 175, 25, 25));

		root.getChildren().add(canvas);
		primaryStage.setScene(new Scene(root));
		primaryStage.show();
	}

}
```

This patch resolved the issue in both the sample code above and in our application where we originally discovered the problem:

[fxgraphics2d_patch.txt](https://github.com/jfree/fxgraphics2d/files/2155903/fxgraphics2d_patch.txt)

**Comment by jfree** [comment]
Good catch, your fix also looks good.

**Comment by jfree** [comment]
Fix will be included in the 1.7 release.
